### PR TITLE
fix(isvtest): parse CUDA UMD Version from nvidia-smi output

### PR DESCRIPTION
**Pull request by abegnoche**
## Summary
- Extend `parse_cuda_version()` to accept both legacy `CUDA Version:` and newer `CUDA UMD Version:` nvidia-smi headers.
- Update `HostSoftwareCheck` and `GpuCheck` to use the shared parser instead of grepping for the legacy header.
- Add unit tests for legacy, UMD, and missing-header cases.

## Test plan
- [x] `uv run python -m pytest isvtest/tests/test_nvidia.py isvtest/tests/test_validation.py::TestHostSoftwareCheckBiosBaselines isvtest/tests/test_validation.py::TestHostSoftwareCheckTpmBaselines -q`

## Summary by CodeRabbit

* **New Features**
  * Added support for detecting CUDA version from newer NVIDIA-SMI output formats alongside legacy formats.

* **Tests**
  * New unit tests for CUDA version parsing covering legacy and modern output formats, version preference handling, and edge cases.

### CODE REVIEW

**Review by mresvanis — APPROVED** [review]

File: isvtest/tests/test_nvidia.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    def test_cuda_umd_version_header(self) -> None:
+        header = "| NVIDIA-SMI 610.47    KMD Version: 610.47    CUDA UMD Version: 13.3     |"
+        assert parse_cuda_version(header) == "13.3"
+
+    def test_prefers_first_match_when_both_present(self) -> None:
+        output = "CUDA Version: 12.4\nCUDA UMD Version: 13.3"
+        assert parse_cuda_version(output) == "12.4"
+
+    def test_returns_none_when_missing(self) -> None:
+        assert parse_cuda_version("Driver Version: 550.54.15") is None
```
[the remark this answers is not in the document]
Comment (abegnoche) [reply isvtest/tests/test_nvidia.py:37]: for tests it's fine, wnf
